libsolidity/interface/CompilerStack.h

```diff
@@ -103,6 +103,9 @@ class CompilerStack: boost::noncopyable
 	/// Sets the given source code as the only source unit apart from standard sources and parses it.
 	/// @returns false on error.
 	bool parse(std::string const& _sourceCode);
+	bool analyze();
```

Comment:
Please add documentation. Can this be called before having called parse? What happens if this does not have any sources set? What does it do?